src/System.Runtime/tests/System/GC.cs

```diff
@@ -46,14 +53,25 @@ private static void VerifyCollectionCount(GCCollectionMode mode)
     }
 
     [Fact]
-    public static void InvalidCollectionModes()
+    public static void TestCollect_Int_GCCollectionMode_Invalid()
     {
-        Assert.Throws<ArgumentOutOfRangeException>(() => GC.Collect(2, (GCCollectionMode)(GCCollectionMode.Default - 1)));
-        Assert.Throws<ArgumentOutOfRangeException>(() => GC.Collect(2, (GCCollectionMode)(GCCollectionMode.Optimized + 1)));
+        Assert.Throws<ArgumentOutOfRangeException>("generation", () => GC.Collect(-1, GCCollectionMode.Default)); // Generation < 0
+
+        Assert.Throws<ArgumentOutOfRangeException>("Enum value was out of legal range.", () => GC.Collect(2, GCCollectionMode.Default - 1)); // Invalid collection mode
```

Comment:
Basically, I think this is a bug. The parameter is meant to be generation or something like that. Here, the message is being passed as the parameter